presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusQueryResponseParse.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.prometheus;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+
+public class PrometheusQueryResponseParse
+{
+    private final InputStream response;
+    private boolean status;
+
+    private String error;
+    private String errorType;
+    private List<String> warnings; //TODO not parsing warnings for now
+    private String resultType;
+    private String result;
+    private List<PrometheusMetricResult> results;
+    private PrometheusTimeSeriesValue stringOrScalarResult;
+
+    public PrometheusQueryResponseParse(InputStream response)
+            throws IOException
+    {
+        this.response = response;
+        parse();
+    }
+
+    private boolean parse()
+            throws IOException
+    {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        JsonParser parser = new JsonFactory().createParser(response);
+        while (!parser.isClosed()) {
+            JsonToken jsonToken = parser.nextToken();
+            if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+                if (parser.getCurrentName().equals("status")) {
+                    parser.nextToken();
+                    if (parser.getValueAsString().equals("success")) {
+                        this.status = true;
+                        while (!parser.isClosed()) {
+                            parser.nextToken();
+                            if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+                                if (parser.getCurrentName().equals("resultType")) {
+                                    parser.nextToken();
+                                    resultType = parser.getValueAsString();
+                                }
+                                if (parser.getCurrentName().equals("result")) {
+                                    parser.nextToken();
+                                    ArrayNode node = mapper.readTree(parser);
+                                    result = node.toString();
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    else {
+                        //error path
```

Comment:
throw prestoexception?

Reply:
Done in 915a0e0f and caught some mistakes made in the error parsing.